### PR TITLE
fix(ui): truncate notice messages if they are too long

### PR DESCRIPTION
**Pull request by monitz87**
**Please make sure the following boxes are checked before submitting your Pull Request:**

- [x] I linked the issues related to this PR in its description (if any).
- [x] If this PR adds new code that is not going to change in the near future, it includes unit tests to cover it.

This PR fixes the notice bar layout being broken if the notice content is too long. It does this by using flexbox to make the bar a bit more responsive and using a javascript library to clamp the text to a maximum of 2 lines (adding ellipses at the end if necessary). There is no pure-CSS way of managing multi-line clamping, so the library is a necessity for now.

### CODE REVIEW

File: packages/polymath-ui/src/components/NoticeBar/NoticeBar.js [hunk 1]
```diff
@@ -3,6 +3,8 @@
 import React, { Component } from 'react';
 import { Icon } from 'carbon-components-react';
 import { connect } from 'react-redux';
+import clamp from 'clamp-js-main';
+import ReactDOMServer from 'react-dom/server';
```
Comment (RafaelVidaurre) [inline packages/polymath-ui/src/components/NoticeBar/NoticeBar.js]: Why was this dependency added?

**Comment by Grsmto** [comment]
Would be sweet to have an .mdx for this (but then we need to extract the Redux part of the component and make it dumb) so we can test it easily.
However if this issue is urgent we can do that later I think.

File: packages/polymath-ui/src/components/NoticeBar/NoticeBar.js [hunk 2]
```diff
@@ -50,24 +52,50 @@ class NoticeBar extends Component<Props> {
...
   }
+
+  componentDidUpdate() {
+    /**
+     * NOTE @monitz87: I had to use this fork of the 'clamp-js' library (https://www.npmjs.com/package/clamp-js-main)
+     * because line clamping is horribly unsupported in CSS (and what little support there is is very browser-specific).
+     * What this code does is truncate the title and paragraph (maintaining the integrity of any HTML elements inside it) if they
+     * exceeds 2 lines, adding ellipses to replace the missing text
+     */
+    const paragraph = document.getElementsByClassName('pui-notice-bar-text')[0];
```
Comment (RafaelVidaurre) [inline packages/polymath-ui/src/components/NoticeBar/NoticeBar.js]: We should use a Ref here instead of a document selector.

Comment (RafaelVidaurre) [reply packages/polymath-ui/src/components/NoticeBar/NoticeBar.js]: Also is there a way we can jut pass this in render? Feels like we are falling out of the React flow

File: packages/polymath-ui/src/components/NoticeBar/NoticeBar.js (hunk 1 above)
Comment (monitz87) [reply packages/polymath-ui/src/components/NoticeBar/NoticeBar.js]:
```suggestion
```

File: packages/polymath-ui/src/components/NoticeBar/NoticeBar.js (hunk 2 above)
Comment (monitz87) [reply packages/polymath-ui/src/components/NoticeBar/NoticeBar.js]: The way clamp works means it has to be done post-render AFAIK. But yeah, refs are a good idea

**Pull request merged**